src/components/molecules/HistoricalChart/HistoricalChart.tsx

```diff
@@ -0,0 +1,353 @@
+            const dataPoint = getCrosshairDataPoint(volumeSeries, param);
+            syncCrosshair(candleStickChart, candlestickSeries, dataPoint);
+            updateLegendData(param);
+        });
+
+        candleStickChart
+            .timeScale()
+            .subscribeVisibleLogicalRangeChange(range => {
+                volumeChart
+                    .timeScale()
+                    .setVisibleLogicalRange(range as LogicalRange);
+            });
+
+        volumeChart.timeScale().subscribeVisibleLogicalRangeChange(range => {
+            candleStickChart
+                .timeScale()
+                .setVisibleLogicalRange(range as LogicalRange);
+        });
+    };
+
+    useEffect(() => {
+        if (!chartContainerRef.current || !secondContainerRef.current) return;
+        const { candlestickSeries, chart: candleStickChart } =
+            createCandlestickChart(chartContainerRef.current);
+
+        const { volumeSeries, chart: volumeChart } = createVolumeChart(
+            secondContainerRef.current,
+            timeScale
+        );
+
+        setupCharts(candlestickSeries, volumeSeries);
+
+        subscribeToChartEvents(
+            candleStickChart,
+            volumeChart,
+            candlestickSeries,
+            volumeSeries
+        );
+
+        volumeChart.timeScale();
+        candleStickChart.timeScale();
+
+        const updateTitleData = (
+            param: MouseEventParams<Time>,
+            sourceChart: string
+        ) => {
+            if (!param.seriesData.size) return;
+
+            let index = 0;
+            if (param.time) {
+                index = candlestickSeries
+                    .data()
+                    .findIndex(
+                        (item: Record<'time', Time>) => item.time === param.time
+                    );
+            }
+
+            const candleData = candlestickSeries.dataByIndex(index);
+            const volumeData = volumeSeries.dataByIndex(index);
+            if (sourceChart === 'candlestick') {
+                syncCrosshair(volumeChart, volumeSeries, candleData);
+            } else {
+                syncCrosshair(candleStickChart, candlestickSeries, volumeData);
+            }
+
+            handleDisplayLegend(candleData, volumeData);
+        };
+
+        candleStickChart.subscribeCrosshairMove(function (param) {
+            updateTitleData(param, 'candlestick');
+        });
+
+        volumeChart.subscribeCrosshairMove(function (param) {
+            updateTitleData(param, 'volume');
+        });
+
+        return () => {
+            candleStickChart.remove();
+            volumeChart.remove();
+            setLegendData({
+                O: '',
+                H: '',
+                L: '',
+                C: '',
+                VOLUME_KEY_NAME: '',
+                Change: '',
+                usdVol: '',
+            });
+            setHoverTime('');
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is there any reason to disable the ts rule here and not adding all deps? We can wrap the functions in useCallabck and avoid this workaround here

Reply:
I fixed this so this is will outdated